[CI] Bump xcode 15.4.0

rmarinho opened the pull request:
### Description of Change

Bumps to use Xcode 15.4.0 and Sonoma machines. 

Device tests  for iOS and Catalyst run on Azure Pipelines macOS-14 , these machines are recreated every time maybe making it more reliable in terms of clean infrastructure. 

This pull request primarily focuses on updating the configuration of the testing environments in the pipeline. The changes involve updating the macOS version from 'Ventura' to 'Sonoma' across several files and updating the required Xcode version from '15.2.0' to '15.4.0'. Additionally, the virtual machine image for iOS and macOS tests has been updated to 'macOS-14'. 

Key changes include:

Updates to macOS version:

* [`eng/pipelines/common/maui-templates.yml`](diffhunk://#diff-beb9b1c0fe7c80ce160eac3a028a267cd8bf6bef41239e0a4bf5829ae3ac25a7L39-R46): Updated the macOS version from 'Ventura' to 'Sonoma' in the parameters for the Android and iOS tests.
* [`eng/pipelines/device-tests.yml`](diffhunk://#diff-d2d1c388a0fb3196dbfcdab96421bc88336637a3d44480c96717f92d000facaeL66-R79): Updated the macOS version from 'Ventura' to 'Sonoma' in the parameters for the Android tests and changed the iOS test parameters to use a new VM pool and image.
* [`eng/pipelines/handlers.yml`](diffhunk://#diff-5c9b6f86c989bb9415ea86f38f9e127f8dadf3a8516e9d027c7ee6289f10c394L104-R111): Updated the macOS version from 'Ventura' to 'Sonoma' in the parameters and stages for the Android and iOS tests. [[1]](diffhunk://#diff-5c9b6f86c989bb9415ea86f38f9e127f8dadf3a8516e9d027c7ee6289f10c394L104-R111) [[2]](diffhunk://#diff-5c9b6f86c989bb9415ea86f38f9e127f8dadf3a8516e9d027c7ee6289f10c394L158-R158) [[3]](diffhunk://#diff-5c9b6f86c989bb9415ea86f38f9e127f8dadf3a8516e9d027c7ee6289f10c394L200-R200) [[4]](diffhunk://#diff-5c9b6f86c989bb9415ea86f38f9e127f8dadf3a8516e9d027c7ee6289f10c394L248-R248)
* [`eng/pipelines/ui-tests.yml`](diffhunk://#diff-01226bae9fe7277cc2f291d47499e45e220b58b560fc0735c13fa6d13e21acd4L70-R70): Updated the macOS version from 'Ventura' to 'Sonoma' in the parameters for the Android and iOS tests, and updated the VM image for macOS tests. [[1]](diffhunk://#diff-01226bae9fe7277cc2f291d47499e45e220b58b560fc0735c13fa6d13e21acd4L70-R70) [[2]](diffhunk://#diff-01226bae9fe7277cc2f291d47499e45e220b58b560fc0735c13fa6d13e21acd4L79-R79) [[3]](diffhunk://#diff-01226bae9fe7277cc2f291d47499e45e220b58b560fc0735c13fa6d13e21acd4L92-R100) [[4]](diffhunk://#diff-01226bae9fe7277cc2f291d47499e45e220b58b560fc0735c13fa6d13e21acd4L109-R109)

Updates to Xcode version:

* [`eng/pipelines/common/variables.yml`](diffhunk://#diff-257fa339ab6d36b7dd4fc3eada944bea777297cec9f90dcd20fbd6b3e6c6f57dL11-R13): Updated the required Xcode version from '15.2.0' to '15.4.0'.

rmarinho committed:
[ios] Bump xcode 15.3.0

jsuarezruiz commented:
```
[12:26:55 VRB] Selected Xcode path: /Applications/Xcode_15.2.0.app
[12:26:55 INF] Xcode Select 15.3.0: ExecutingAction -> NotProvisioned
[12:26:55 ERR] Error
Xamarin.Provisioning.ProvisioningException: Failed to provision /Applications/Xcode_15.3.0.app
```

rmarinho commented:
Trying a run using macOS-14 of Azure Pipelines for:

DeviceTests https://dev.azure.com/xamarin/public/_build/results?buildId=116399&view=results

UITests https://dev.azure.com/xamarin/public/_build/results?buildId=116400&view=results

rmarinho committed:
Move to sonoma and XCode 15.4.0

Move uitests to Sonoma too

Fix pack

Move to macOS-14

Move also device tests catalyst

try use azure pipelines

Dont use azure pipelines

[ios] Move to XCode 15.4

Can android run on sonoma only

Try not cleanup on iOS Azure pipelines

Try again skip cleanup

Fix

Need to provision the certs

Fix

Fix

rmarinho commented:
/rebase
